ast/schema_test.go

```diff
@@ -16,12 +18,15 @@ func testParseSchema(t *testing.T, schema string, expectedType types.Type) {
 		t.Fatalf("unexpected error: %s", err)
 	}
 	newtype, err := loadSchema(sch)
-	if err != nil {
+	if err != nil && !strings.Contains(err.Error(), expectedError.Error()) {
```

Comment:
ℹ️ Comparing errors by looking at their strings is a bit discouraged. I suppose it would be nice to introduce error types and can be inspected instead, using the `error.Is` and `error.As` helpers that golang 1.13 introduced... 

(Not a blocker for this PR, I think.)

Reply:
I changed the error comparison from comparing strings to comparing them through error.Is(err, expectedError). Is this along the lines of what you were looking for, or does "introduce error types" mean that I should do something more? Also, all of your other comments have been addressed. :)